### PR TITLE
bpo-39159: Declare errors that might be raised from literal_eval

### PR DESCRIPTION
**Pull request by isidentical**
https://bugs.python.org/issue39159

### CODE REVIEW

File: Doc/library/ast.rst [hunk 1]
```diff
@@ -1563,7 +1563,7 @@ and classes for traversing abstract syntax trees:
    Safely evaluate an expression node or a string containing a Python literal or
    container display.  The string or node provided may only consist of the
    following Python literal structures: strings, bytes, numbers, tuples, lists,
-   dicts, sets, booleans, and ``None``.
+   dicts, sets, booleans, ``None`` and ``Ellipsis``.
```
Comment (rhettinger) [inline Doc/library/ast.rst:1566]: Let's also take this opportunity to be more specific about what is meant by "numbers".  Replace that  with "ints, floats, complex numbers".

Also, please make the same update to the docstring for *ast.literal_eval()*.

File: Doc/library/ast.rst [hunk 2]
```diff
@@ -1575,6 +1575,10 @@ and classes for traversing abstract syntax trees:
       sufficiently large/complex string due to stack depth limitations
       in Python's AST compiler.
 
+      It can raise :exc:`ValueError`, :exc:`TypeError`, :exc:`SyntaxError`,
+      :exc:`MemoryError` and :exc:`RecursionError` depending on the malformed
+      input.
+
```
Comment (rhettinger) [inline Doc/library/ast.rst:1581]: Just for my edification and for future reference, can you add a non-displayed in-line documentation comment showing an example of how each of these could happen.

Comment (isidentical) [reply Doc/library/ast.rst:1581]:
```py
import ast

recursion_tuple = ast.Tuple(elts=[], ctx=ast.Load())
recursion_tuple.elts.append(recursion_tuple)
INPUTS = [
    ("invalid syntax", SyntaxError),
    ("test", ValueError),
    (ast.List(elts=None, ctx=ast.Load()), TypeError),
    (recursion_tuple, RecursionError),
]

for inp, exc_type in INPUTS:
    try:
        ast.literal_eval(inp)
    except exc_type:
        continue
    else:
        assert False
```
does something like this suits?

Comment (isidentical) [reply Doc/library/ast.rst:1581]: @rhettinger or maybe we can put this in an actual test in `test_ast` and refer it? How does that sounds?

**Comment by taleinat** [comment]
Ping, @rhettinger?

**Comment by rhettinger** [comment]
Triggering a rebuild.

**Pull request merged**